docs: update EKS auth options with IRSA and instance role options (#2962)
elastic/security-docs: docs/cloud-native-security/get-started-with-kspm.asciidoc

```
====

[discrete]
[[kspm-use-irsa]]
===== Option 1 - [Recommended] Use Kubernetes Service Account to assume IAM role

> Levarege IRSA (IAM role to Kubenretes service-account) feature to get temporary credentials and scoped permission.

Reviewer: ```suggestion
Leverage IRSA (IAM role to Kubernetes service-account) feature to get temporary credentials and scoped permission.
```

From https://aws.github.io/aws-eks-best-practices/security/docs/iam/#iam-roles-for-service-accounts-irsa[EKS Best Practices]:

> > IRSA is a feature that allows you to assign an IAM role to a Kubernetes service account. It works by leveraging a Kubernetes feature known as Service Account Token Volume Projection. When Pods are configured with a Service Account that references an IAM Role, the Kubernetes API server will call the public OIDC discovery endpoint for the cluster on startup. The endpoint cryptographically signs the OIDC token issued by Kubernetes and the resulting token mounted as a volume. This signed token allows the Pod to call the AWS APIs associated IAM role. When an AWS API is invoked, the AWS SDKs calls sts:AssumeRoleWithWebIdentity. After validating the token's signature, IAM exchanges the Kubernetes issued token for a temporary AWS role credential.

Reviewer: ```suggestion
> IRSA is a feature that allows you to assign an IAM role to a Kubernetes service account. It works by leveraging a Kubernetes feature known as Service Account Token Volume Projection. When pods are configured with a Service Account that references an IAM Role, the Kubernetes API server will call the public OIDC discovery endpoint for the cluster on startup. The endpoint cryptographically signs the OIDC token issued by Kubernetes and the resulting token is mounted as a volume. This signed token allows the pod to call the AWS API's associated IAM role. When an AWS API is invoked, the AWS SDK calls `sts:AssumeRoleWithWebIdentity`. After validating the token's signature, IAM exchanges the Kubernetes issued token for a temporary AWS role credential.
```

> To use this, do not fill any option in the "Setup Access" section. Click on "Save and contiune" and follow the https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html[IRSA] guide.

Reviewer: ```suggestion
To use this, do not fill in any part of the "Setup Access" section. Click *Save and continue*, and follow the https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html[IRSA] guide.
```

[discrete]
[[kspm-use-instance-role]]
===== Option 2 - Use default instance role
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html[IAM roles for Amazon EC2]

> When you create an IAM role using the IAM console, the console creates an instance profile automatically and gives it the same name as the role to which it corresponds.

> To use this, do not fill any option in the "Setup Access" section. Click on "Save and contiune".

Reviewer: ```suggestion
To use this, do not fill in any part of the "Setup Access" section. Click *Save and continue*.
```

----

To use this authentication method, leave all options empty and follow the https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html[IRSA] guide.

[discrete]
[[kspm-use-instance-role]]
===== Option 2 - Use default instance role
From https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/iam-roles-for-amazon-ec2.html[IAM roles for Amazon EC2]:

> When you create an IAM role using the IAM console, the console creates an instance profile automatically and gives it the same name as the role to which it corresponds.

> To use this authentication method, leave all the options empty.

Reviewer: Leave all options empty while doing what? While creating the IAM role whose instance profile you're going to use? A specific link to a page that says exactly what to do would be helpful here.

Author: Thank you, @benironside.
I have updated the description.
@tinnytintin10 we might want to expose the IRSA and the Assigned Role in the UI.

Reviewer 2: @olegsu quick clarification, would we only be aware of what the IRSA and assigned role are after elastic-agent has been deployed?

Author: We can assume that either IRSA or an assigned role will be used when the form was left empty (no arn, no creds, etc).

> This is important to validate that the assigned IAM role has all the required permissions.

Reviewer: What is important to validating the permissions? To leave all options empty while setting up the IAM role? Can you please give a little more context here?
